include/lbann/utils/conduit_extensions.hpp

```diff
@@ -0,0 +1,45 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2023, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// http://github.com/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+
+#include "conduit/conduit_data_array.hpp"
+
+namespace conduit {
+
+template <typename T>
+T
+data_array_prod(DataArray<T> a)
+{
+    T res = 1;
+    for(index_t i = 0; i < a.number_of_elements(); i++)
+    {
+        const T &val = a.element(i);
+        res *= val;
+    }
+
+    return res;
+}
+
+} // conduit
```

Comment:
```suggestion
namespace lbann {

template <typename T>
T
data_array_prod(DataArray<T> const& a)
{
    T res = 1;
    for(index_t i = 0; i < a.number_of_elements(); i++)
    {
        const T &val = a.element(i);
        res *= val;
    }

    return res;
}

} // namespace lbann
```
IMO we shouldn't be extending their namespace; just put it in lbann. Also, conduit::DataArray doesn't own any data, so it's just a "shallow copy" (pointer and size), but the "expected" API would probably be pass-by-const-reference. (I'm moderately surprised that they don't expose begin() and end() functions (and suitable iterators) for this container!)